cmd/extensions-validator/cmd/validate.go

```diff
@@ -86,6 +90,25 @@ func validateRootfs() error {
 		return fmt.Errorf("invalid version format %s for extension: %s", extension.Manifest.Metadata.Version, extension.Manifest.Metadata.Name)
 	}
 
+	// find all yaml files in usr/local/etc/containers
+	extensionServiceSpecFiles, err := filepath.Glob(filepath.Join(extension.RootfsPath(), "usr/local/etc/containers", "*.yaml"))
+	if err != nil {
+		return fmt.Errorf("error finding service spec files: %w", err)
+	}
+
+	for _, extensionServiceSpecFile := range extensionServiceSpecFiles {
+		var extensionServiceSpec services.Spec
+
+		extensionServiceSpecData, err := os.ReadFile(extensionServiceSpecFile)
+		if err != nil {
+			return fmt.Errorf("error reading service spec file %s: %w", extensionServiceSpecFile, err)
+		}
+
+		if err := yaml.Unmarshal(extensionServiceSpecData, &extensionServiceSpec); err != nil {
```

Comment:
we can also do validate there: https://github.com/siderolabs/talos/blob/85110deec25d0fbfe1cdfc4eec2189ad48bec8ad/pkg/machinery/extensions/services/services.go#L88-L90